pkg/aws/bucket.go

```diff
@@ -42,22 +42,27 @@ import (
 
 type bucket struct {
 	*config.Bucket
-	storage *storage
+	account *accountProvider
 	s3      *s3.S3
 
+	storage *storage
+
 	access string
 	bucket *s3.Bucket
+	arn    string
 }
 
-func newBucket(bkt resource.Bucket, cfg *config.Bucket, s *s3.S3) (resource.ProviderBucket, error) {
+func newBucket(bkt resource.Bucket, cfg *config.Bucket, s *s3.S3, acc *accountProvider) (resource.ProviderBucket, error) {
```

Comment:
I would prefer "prov" over "acc" here.

See previous comment about parameter list, the *s3.S3 parameter isn't needed since cfg and provider are being passed in.